### PR TITLE
fixes returning empty files

### PR DESCRIPTION
**Pull request by pmbohm**
Fixes https://github.com/aodn/aodn-portal/issues/2268

### CODE REVIEW

File: src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java [hunk 1]
```diff
@@ -409,25 +405,37 @@ public static int execute(List<String> command) throws Exception {
...
             String line;
+
             while ((line = br.readLine()) != null) {
                 logger.info(line);
+                bashLog.append(line);
             }
 
             process.waitFor();
 
+            if (process.exitValue() != 0) {
```
Comment (anguss00) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java]: I'm a bit concerned about the handling here of process. In my testing, the exception is thrown and bubbled up but by the time we have an exit value for the process, the damage has already been done and the file has been created (and therefore returned). So we get an exception in our catalina log. So, my suggestion would be to consider setting the processListener.isCancelled(true) whenever we throw a GGD exception. Needs testing and validation of course.

Comment (bhasin85) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java]: @anguss00  I am able to reproduce the issue using the portal. I am updating the fix.

Comment (anguss00) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java]: Ok, cool. Thanks @bhasin85

Comment (bhasin85) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java]: @anguss00  I tried replicating the [error](https://github.com/aodn/aodn-portal/issues/2263) with my changes without any success. I believe the exception handling will take care of it as earlier the errors were getting logged as info and some exceptions were suppressed. Let me know if you have any concern/question.

Comment (pmbohm) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java]: `processListener.isCancelled(true) ` worked well for me. I believed you were going to expand on this in other functions. As @anguss00 commented previously  `to consider setting the processListener.isCancelled(true) whenever we throw a GGD exception`

Comment (bhasin85) [inline src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java]: @pmbohm  @anguss00 

> To reproduce
> 
> Perform a subset where an ncks, npdqd or other command fails e.g. #2263

I think the changes fixes the issue.

**Pull request merged**